### PR TITLE
Pass media element

### PR DESCRIPTION
**Pull request by AxelDelmas**
Pass media element to peer-agent module

### CODE REVIEW

File: lib/DashjsWrapperPrivate.js [hunk 1]
```diff
@@ -80,6 +80,14 @@ class DashjsWrapperPrivate {
...
             INTEGRATION_VERSION
         );
+
+        try {
+            let mediaElement = this._player.getVideoElement(); // Throws if media element is not attached, thus the try / catch block
+            this._peerAgentModule.setMediaElement(mediaElement);
+        } catch (e) {
+            // There is no event on dash.js notifying the moment where the video is attached: simply warn in the console
+            console.warn("Tried to access media element before it was set");
+        }
```
Comment (tchakabam) [inline lib/DashjsWrapperPrivate.js]: that seems a bit fuzzy. in what cases will `getVideoElement` throw?

Comment (tchakabam) [inline lib/DashjsWrapperPrivate.js]: i.e when is this function running before the video is attached. and how could we avoid it?

Comment (AxelDelmas) [inline lib/DashjsWrapperPrivate.js]: We can't. Dash.js API authorizes the user to attach the media and to attach the source separately. Our module loads when the manifest is loaded, so if the user attach the source first, and waits a while before attaching the video (for example pre-loading the content while an ad is playing), `getVideoElement` will throw in the meantime

**Comment by tchakabam** [comment]
👍

**Pull request merged**